src/TritonCTS/src/HTreeBuilder.cpp

```diff
@@ -287,47 +285,45 @@ void HTreeBuilder::computeLevelTopology(unsigned level, double width, double hei
         for (unsigned charSegLength = _techChar->getMaxSegmentLength(); charSegLength >= 1; --charSegLength) {
                 unsigned numWires = (segmentLength - currLength) / charSegLength;
                 
-                if (numWires < 1) {
-                        continue;
-                }
-
-                currLength += numWires * charSegLength;
-                for (unsigned wireCount = 0; wireCount < numWires; ++wireCount) {
-                        unsigned outCap = 0, outSlew = 0;
-                        unsigned key = 0;
-                        if (_options->isSimpleSegmentEnabled()){
-                                remainingLength = remainingLength - charSegLength;
-                                
-                                if (segmentLength >= vertexBufferLength && (wireCount + 1 >= numWires) && _options->isVertexBuffersEnabled()){
-                                        remainingLength = 0;
-                                        key = computeMinDelaySegment(charSegLength, inputSlew, inputCap, 
-                                                                SLEW_THRESHOLD, INIT_TOLERANCE, outSlew, outCap, true, remainingLength);
-                                        remainingLength = remainingLength + _options->getBufferDistance() / (_techChar->getLengthUnit());
-                                } else {
-                                        if (remainingLength <= 0){
+                if (numWires >= 1) {
+                        currLength += numWires * charSegLength;
+                        for (unsigned wireCount = 0; wireCount < numWires; ++wireCount) {
```

Comment:
unsigned can bite you pretty hard if you ever get a negative result doing a subtraction. safer to use int

Reply:
I've removed a bunch of unsigned variables that were related to important data (such as here and the characterization). Some values are still saved in the CTS structures as unsigned (looking into the most important ones to change).